java-checks/src/main/java/org/sonar/java/checks/HardcodedURICheck.java

```diff
@@ -179,6 +179,8 @@ private void checkVariable(VariableTree tree) {
 
     String stringLiteral = stringLiteral(initializer);
     if (stringLiteral == null) {
+      // the visitor only report path delimiter on binary expressions, so the initializer must not be a string literal
```

Comment:
It took me a while to understand the meaning of the comment. I suggest changing it, for example, "The check below applies only to expression, so we need it only if the initializer is not a literal."